docs/data/data-grid/server-side-data/ServerSideRowGroupingExpansionPersistence.tsx

```diff
@@ -0,0 +1,87 @@
+import * as React from 'react';
+import {
+  DataGridPremium,
+  GridDataSource,
+  useGridApiRef,
+  useKeepGroupedColumnsHidden,
+  GridGetRowsResponse,
+  GRID_ROOT_GROUP_ID,
+} from '@mui/x-data-grid-premium';
+import { useMockServer } from '@mui/x-data-grid-generator';
+import Button from '@mui/material/Button';
+
+export default function ServerSideRowGroupingExpansionPersistence() {
+  const apiRef = useGridApiRef();
+
+  const { fetchRows, columns } = useMockServer<GridGetRowsResponse>(
+    { dataSet: 'Movies' },
+    {},
+  );
+
+  const dataSource: GridDataSource = React.useMemo(() => {
+    return {
+      getRows: async (params) => {
+        const urlParams = new URLSearchParams({
+          paginationModel: JSON.stringify(params.paginationModel),
+          filterModel: JSON.stringify(params.filterModel),
+          sortModel: JSON.stringify(params.sortModel),
+          groupKeys: JSON.stringify(params.groupKeys),
+          groupFields: JSON.stringify(params.groupFields),
+        });
+        const getRowsResponse = await fetchRows(
+          `https://mui.com/x/api/data-grid?${urlParams.toString()}`,
+        );
+        return {
+          rows: getRowsResponse.rows,
+          rowCount: getRowsResponse.rowCount,
+        };
+      },
+      getGroupKey: (row) => row.group,
+      getChildrenCount: (row) => row.descendantCount,
+    };
+  }, [fetchRows]);
+
+  const initialState = useKeepGroupedColumnsHidden({
+    apiRef,
+    initialState: {
+      rowGrouping: {
+        model: ['company', 'director'],
+      },
+    },
+  });
+
+  return (
+    <div style={{ width: '100%' }}>
+      <div style={{ display: 'flex', gap: 8, paddingBottom: 4 }}>
+        <Button
+          variant="outlined"
+          onClick={() => {
+            apiRef.current?.dataSource.fetchRows();
```

Comment:
Side note: is there a way to refetch all rows (including nested expanded groups) in the background while keeping the expansion and scrolling state?